src/main/java/frc/robot/commands/ScoreCoralCommand.java

```diff
@@ -0,0 +1,120 @@
+package frc.robot.commands;
+
+import com.ctre.phoenix6.swerve.SwerveRequest;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.ArmSubsystem;
+import frc.robot.subsystems.CommandSwerveDrivetrain;
+import frc.robot.subsystems.GamePieceManipulatorSubsystem;
+import java.util.function.Supplier;
+
+/**
+ * A GamePieceManipulatorCommands uses the GamepieceMainpulatorSubsytems to score the coral.
+ */
+public class ScoreCoralCommand extends Command {
+  // The Subsystem the command runs on.
+  private final GamePieceManipulatorSubsystem gamePieceManipulatorSubsystem;
+  private final ArmSubsystem armSubsystem;
+  private final CommandSwerveDrivetrain swerveDrivetrain;
+
+  private final Supplier<Rotation2d> robotAngleSupplier;
+  private final Supplier<Pose2d> poseSupplier;
+  private final Supplier<ChassisSpeeds> chassSupplier;
+  private final int levelToScore;
+
+  /**
+   * The construcnter of the command and tell the perimeteres.
+   * 
+   * @param manipulator
+   * @param arm
+   * @param drive
+   * @param poseSupplier
+   * @param robotAngleSupplier
+   * @param chassSupplier
+   * @param levelToScore
+   */
+  public ScoreCoralCommand(
+      GamePieceManipulatorSubsystem manipulator,
+      ArmSubsystem arm,
+      CommandSwerveDrivetrain drive,
+      Supplier<Pose2d> poseSupplier,
+      Supplier<Rotation2d> robotAngleSupplier,
+      Supplier<ChassisSpeeds> chassSupplier,
+      int levelToScore) {
+    this.gamePieceManipulatorSubsystem = manipulator;
+    this.armSubsystem = arm;
+    this.levelToScore = levelToScore;
+    this.swerveDrivetrain = drive;
+    addRequirements(swerveDrivetrain, armSubsystem, gamePieceManipulatorSubsystem);
+
+    this.poseSupplier = poseSupplier;
+    this.robotAngleSupplier = robotAngleSupplier;
+    this.chassSupplier = chassSupplier;
+  }
+
+  /**
+   * Allows the robot to decide what elave it should go to.
+   */
+  @Override
+  public void initialize() {
+    var robotPose = poseSupplier.get();
+    var robotAngle = robotAngleSupplier.get();
+    final SwerveRequest.FieldCentric _drive = new SwerveRequest.FieldCentric();
+    var fieldSpeeds = new Translation2d().rotateBy(robotAngle);
```

Comment:
Not sure what this is trying to do. It's rotating (0, 0) by the angle of the robot, but rotating (0, 0) will always be (0, 0)